### PR TITLE
Nexus 9835 assets

### PR DESCRIPTION
**Pull request by mosabua**
Just a short section to explain the concepts around what an "asset" is and how it relates to the components. This came up as an issue from some user feedback.

Issue: https://issues.sonatype.org/browse/NEXUS-9835

CI plan with rendered results: http://bamboo.s/browse/BOOK-NEXUS97-1

### CODE REVIEW

File: chapter-concepts.asciidoc [hunk 1]
```diff
@@ -42,9 +42,24 @@ Components provide all the building blocks and features that allow a development
...
 
+.Assets
+
+Components are an abstraction identified by a set of specific values - the 'coordinates'. A generic set of these
+coordinates is the usage of 'group', 'name' and 'version'. The names and the usage of these coordinates changes
+with the tool-chains used. Components can also be the anchor for further meta data. Assets are the material
+addition to all this meta data. The actual archive file is an asset associated to the component. Many formats have
+a one to one mapping for component to asset.
+
+More complex formats however have numerous assets associated to a component. For example a typical JAR component
```
Comment (mrprescott) [inline chapter-concepts.asciidoc]: associated *with*

File: chapter-concepts.asciidoc [hunk 2]
```diff
@@ -42,9 +42,24 @@ Components provide all the building blocks and features that allow a development
 applications by assembling them and adding their own business related components to create a full-fledged,
 powerful application.
 
-In different toolchains components are called 'artifact', 'package', 'bundle', 'archive' and other terms. The
+In different tool-chains components are called 'artifact', 'package', 'bundle', 'archive' and other terms. The
 concept and idea remains the same and 'component' is used as the independent, generic term.
 
+.Assets
+
+Components are an abstraction identified by a set of specific values - the 'coordinates'. A generic set of these
```
Comment (mrprescott) [inline chapter-concepts.asciidoc]: 'repeatability' is one word, I believe; same with 'metadata'

File: chapter-concepts.asciidoc [hunk 3]
```diff
@@ -42,9 +42,24 @@ Components provide all the building blocks and features that allow a development
...
+coordinates is the usage of 'group', 'name' and 'version'. The names and the usage of these coordinates changes
+with the tool-chains used. Components can also be the anchor for further meta data. Assets are the material
+addition to all this meta data. The actual archive file is an asset associated to the component. Many formats have
+a one to one mapping for component to asset.
```
Comment (mrprescott) [inline chapter-concepts.asciidoc]: one-to-one mapping

File: chapter-concepts.asciidoc (hunk 2 above)
Comment (mosabua) [inline chapter-concepts.asciidoc]: Hahah.. both came from spell check. I checked.. both are valid.. I will go back to your choice ;-)

File: chapter-concepts.asciidoc [hunk 4]
```diff
@@ -16,7 +16,7 @@ The {platform} with {pro}, {oss} and {iq} is all about working with components a
 .So what are components?
 
 A component is a resource like a library or a framework that is used as part of your software application at
-runtime, integration or unit test execution time or required as part of your build process. It can also be an
+run-time, integration or unit test execution time or required as part of your build process. It can also be an
```
Comment (mrprescott) [inline chapter-concepts.asciidoc]: can also -> could
-without any dynamic behavior- (I think it's clearer without that bit.)

File: chapter-concepts.asciidoc (hunk 1 above)
Comment (mosabua) [inline chapter-concepts.asciidoc]: Could be either.. but yes "with" is more common and correct. Changed.

**Comment by mosabua** [comment]
@mrprescott  all good otherwise? +1 maybe?

File: chapter-concepts.asciidoc (hunk 3 above)
Comment (mosabua) [inline chapter-concepts.asciidoc]: fixed

File: chapter-concepts.asciidoc (hunk 4 above)
Comment (mosabua) [inline chapter-concepts.asciidoc]: Fixed

File: chapter-concepts.asciidoc [hunk 5]
```diff
@@ -42,9 +42,24 @@ Components provide all the building blocks and features that allow a development
...
 powerful application.
 
-In different toolchains components are called 'artifact', 'package', 'bundle', 'archive' and other terms. The
+In different tool-chains components are called 'artifact', 'package', 'bundle', 'archive' and other terms. The
 concept and idea remains the same and 'component' is used as the independent, generic term.
 
+.Assets
+
+Components are an abstraction identified by a set of specific values - the 'coordinates'. A generic set of these
+coordinates is the usage of 'group', 'name' and 'version'. The names and the usage of these coordinates changes
```
Comment (mrprescott) [inline chapter-concepts.asciidoc]: (This section is slightly odd in that it seems to start off by defining components, which I think you've already done well enough above. I'd move the explanation of component coordinates into the 'components' section, or its own section.)

Components are ~an abstraction~ identified by a set of values - their 'coordinates'.

File: chapter-concepts.asciidoc [hunk 6]
```diff
@@ -194,7 +209,7 @@ standards and procedures for deploying components.  Much like the difference bet
 staging network, a release repository is considered a production network and a snapshot repository is more like a
 development or a testing network.  While there is a higher level of procedure and ceremony associated with
 deploying to a release repository, snapshot components can be deployed and changed frequently without regard for
-stability and repeatability concerns.
+stability and repeat-ability concerns.
```
Comment (mrprescott) [inline chapter-concepts.asciidoc]: All these new hyphens look odd to me. :P

File: chapter-concepts.asciidoc [hunk 7]
```diff
@@ -42,9 +42,24 @@ Components provide all the building blocks and features that allow a development
...
+Components are an abstraction identified by a set of specific values - the 'coordinates'. A generic set of these
+coordinates is the usage of 'group', 'name' and 'version'. The names and the usage of these coordinates changes
+with the tool-chains used. Components can also be the anchor for further meta data. Assets are the material
+addition to all this meta data. The actual archive file is an asset associated to the component. Many formats have
+a one to one mapping for component to asset.
+
+More complex formats however have numerous assets associated to a component. For example a typical JAR component
+in a Maven repository is defined at least by the POM and the JAR files - both of which constitute separate assets
+belonging to the same components. Additional files such as JavaDoc or Sources JAR files are assets that belong to
+the same component. The Docker format on the other hand gives assets unique identifier and allows for reuse of
```
Comment (mrprescott) [inline chapter-concepts.asciidoc]: The Docker format, on the other hand, gives assets unique identifers ..

File: chapter-concepts.asciidoc [hunk 8]
```diff
@@ -42,9 +42,24 @@ Components provide all the building blocks and features that allow a development
...
+them between different components. The so-called layer is for example a specific operating system and multiple
```
Comment (mrprescott) [inline chapter-concepts.asciidoc]: "The so-called layer is for example" is quite awkward - how about, "A Docker layer, for example, could be a specific operating system, referenced by multiple Docker images."

**Comment by mrprescott** [comment]
+1

File: chapter-concepts.asciidoc (hunk 6 above)
Comment (mosabua) [inline chapter-concepts.asciidoc]: Fixed

File: chapter-concepts.asciidoc (hunk 5 above)
Comment (mosabua) [inline chapter-concepts.asciidoc]: Moved up.

File: chapter-concepts.asciidoc (hunk 7 above)
Comment (mosabua) [inline chapter-concepts.asciidoc]: Added ;-)

File: chapter-concepts.asciidoc (hunk 8 above)
Comment (mosabua) [inline chapter-concepts.asciidoc]: Rejigged..

**Comment by mosabua** [comment]
Thanks for the rocking review @mrprescott - now I can merge and close another issue this week ;-)

**Pull request merged**